src/main/java/com/svydovets/bibirnate/utils/ValidationUtils.java

```diff
@@ -0,0 +1,26 @@
+package com.svydovets.bibirnate.utils;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+
+import com.svydovets.bibirnate.configuration.NotNull;
+import com.svydovets.bibirnate.exceptions.PropertiesFileValidationException;
+
+public class ValidationUtils {
+
+    public static <T> void validateNotNullFields(Class<T> entityType, T instance) {
+        List<Field> collect = Arrays.stream(entityType.getDeclaredFields()).peek(f -> f.setAccessible(true))
+                .filter(f -> {
+                    try {
+                        return f.isAnnotationPresent(NotNull.class) && f.get(instance) == null;
+                    } catch (IllegalAccessException ex) {
+                        throw new PropertiesFileValidationException(ex);
+                    }
```

Comment:
I prefer to keep the code with the Stream API as short as it is possible, we can move this logic to a method.
Also, as far as I remember, peek should be used only for debugging, and it doesn't guarantee that it will handle all items inside stream.

Reply:
Moved to separate method